src/sentry/rules/processing/delayed_processing.py

```diff
@@ -200,38 +203,90 @@ def parse_rulegroup_to_event_data(
     return parsed_rulegroup_to_event_data
 
 
-def get_group_to_groupevent(
+def bulk_fetch_events(event_ids: list[str], project_id: int) -> dict[str, Event]:
+    event_limit = 100
+    node_id_to_event_id: dict[str, str] = {}
+
+    for event_id_chunk in chunked(event_ids, event_limit):
+        for event_id in event_id_chunk:
+            node_id = Event.generate_node_id(project_id, event_id=event_id)
+            node_id_to_event_id[node_id] = event_id
+
+    node_ids = list(node_id_to_event_id.keys())
+    fetch_retry_policy = ConditionalRetryPolicy(should_retry_fetch, exponential_delay(1.00))
+    bulk_data = fetch_retry_policy(lambda: nodestore.backend.get_multi(node_ids))
```

Comment:
This is the part that needs to be batched, right now you're still fetching all the node ids at the same time

Reply:
Oh oops lol, hopefully batched the right thing now